### PR TITLE
Move Types

### PR DESCRIPTION
**Pull request by billyjbryant**
## Description
Adds support for Move Types to the move data files.  Also adds in Move Type DTS fields. 

### DTS 
The following DTS fields have been added to both **Monster** and **Raid** alerts

| DTS              | Description                                             |
|----------------- |-------------------------------------------------------- |
| quick_type_id | The id of the Quick Move's Type or  `?`.          |
| quick_type     | The name of the Quick Move's Type or `Unknown`.        |
| quick_type_emoji            | The emoji for the Quick Move's Type or Empty.        |
| charge_type_id | The id of the Charge Move's Type or  `?`.          |
| charge_type     | The name of the Charge Move's Type or `Unknown`.        |
| charge_type_emoji            | The emoji for the Charge Move's Type or Empty.        |

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (would cause existing functionality to change)

## Motivation and Context
Knowing the type of a move for a monster or raid boss is beneficial when planning raid groups and building effective battle strategies.  

## How Has This Been Tested?
This was tested on my production instance

## Screenshots (if appropriate):
image

image

## Wiki Update

- [x] This change requires an update to the Wiki.
- [ ] This change does not require an update to the Wiki.

## Checklist
- [x] This change follows the code style of this project.
- [x] This change only changes what is necessary to the feature.

### CODE REVIEW

**Comment by billyjbryant** [comment]
I found an issue with this, it looks to be using the same ID for both.

**Comment by billyjbryant** [comment]
Fixed the issue with the Emoji and provided better screenshots.

File: data/move_info.json [hunk 1]
```diff
@@ -1,1115 +1,1301 @@
 {
   "013": {
+	"type": 1,
```
Comment (MARTINI1) [inline data/move_info.json]: tabulate issue

File: data/move_info.json [hunk 2]
```diff
@@ -1,1115 +1,1301 @@
...
     "damage": 60,
     "dps": 20.69,
     "duration": 2900,
     "energy": 33
-  },
+  },
```
Comment (MARTINI1) [inline data/move_info.json]: another tabulate that is left over

File: PokeAlarm/Events/MonEvent.py [hunk 3]
```diff
@@ -196,16 +199,24 @@ def generate_dts(self, locale, timezone, units):
             'form_id_3': "{:03d}".format(self.form_id),
 
             # Quick Move
-            'quick_move': locale.get_move_name(self.quick_move_id),
-            'quick_id': self.quick_move_id,
+            'quick_move': locale.get_move_name(self.quick_id),
+            'quick_id': self.quick_id,
+            'quick_type_id': self.quick_type,
+            'quick_type': locale.get_type_name(self.quick_type),
+            'quick_type_emoji': Unknown.or_empty(
```
Comment (kvangent) [inline PokeAlarm/Events/MonEvent.py]: Doesn't type emoji return `''`? If so, the unknown check is superfluous

File: PokeAlarm/Events/MonEvent.py [hunk 4]
```diff
@@ -196,16 +199,24 @@ def generate_dts(self, locale, timezone, units):
...
             'quick_energy': self.quick_energy,
 
             # Charge Move
-            'charge_move': locale.get_move_name(self.charge_move_id),
-            'charge_id': self.charge_move_id,
+            'charge_move': locale.get_move_name(self.charge_id),
+            'charge_id': self.charge_id,
+            'charge_type_id': self.charge_type,
+            'charge_type': locale.get_type_name(self.charge_type),
+            'charge_type_emoji': Unknown.or_empty(
```
Comment (kvangent) [inline PokeAlarm/Events/MonEvent.py]: Same about Unknown check

File: PokeAlarm/Utils.py [hunk 5]
```diff
@@ -128,6 +128,18 @@ def get_team_id(team_name):
...
+# Returns the types of a move when requesting
+def get_move_type(move_id):
+    if not hasattr(get_move_type, 'info'):
+        get_move_type.info = {}
+        file_ = get_path('data/move_info.json')
+        with open(file_, 'r') as f:
+            j = json.loads(f.read())
+        for id_ in j:
+            get_move_type.info[int(id_)] = j[id_]['type']
+    return get_move_type.info.get(move_id, 'unkn')
```
Comment (kvangent) [inline PokeAlarm/Utils.py]: We should hardcode this Unknown using the unknown class

Comment (billyjbryant) [reply PokeAlarm/Utils.py]: Do you want me to do the same for the rest of the Move Data while I'm in here or submit that via another PR?

Comment (kvangent) [reply PokeAlarm/Utils.py]: That can be in another PR - I think that we need to change our locales change to a singleton now that we removed multiprocessing. There's no reason to keep an identical copy of the mappings per manager.

Comment (billyjbryant) [reply PokeAlarm/Utils.py]: I also think a lot of the functions in `Utils.py` really need to be moved to `MonUtils.py` so that can be another cleanup PR

**Pull request merged**